### PR TITLE
chore(styleguide): used decoratedStory for knobs.escapeHTML in Storybook

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## Used decoratedStory for knobs.escapeHTML in Storybook

Fixes WEB-802

1. Renames the `decoratedStory` function to `decoratedStories`, so it then:
2. Adds a new `decoratedStory` function for all stories that enables escaping HTML

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e90ded3973e3c9da6a9ff8d--codecademy-styleguide.netlify.com>https://5e90ded3973e3c9da6a9ff8d--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e90ded3973e3c9da6a9ff8d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e90e6cb1694d9de27a57053--codecademy-styleguide.netlify.com>https://5e90e6cb1694d9de27a57053--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e90e6cb1694d9de27a57053>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e950e04fda81b1d905ec1d0--codecademy-styleguide.netlify.com>https://5e950e04fda81b1d905ec1d0--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e950e04fda81b1d905ec1d0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e9511094bae0d4a7a591f87--codecademy-styleguide.netlify.com>https://5e9511094bae0d4a7a591f87--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e9511094bae0d4a7a591f87>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e9dc069d59141026daf0b2f--codecademy-styleguide.netlify.app>https://5e9dc069d59141026daf0b2f--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e9dc069d59141026daf0b2f>Deploy Logs</a></p>

File: packages/styleguide/stories/Templating/DecoratedStory.ts [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
-export type Section =
-  | 'Atoms'
-  | 'Brand'
-  | 'Foundations'
-  | 'Molecules'
-  | 'Organisms';
+type DecoratedStory = {
+  (name: string, context: () => void): any;
+  (context: () => void): any;
+};
```
Comment (codecaaron) [inline packages/styleguide/stories/Templating/DecoratedStory.ts:4]: The argument reordering here might be a bit confusing for these overloads, but I see how it matches the other syntax

File: packages/styleguide/stories/Templating/DecoratedStory.ts [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
 
-const stringify = (component: string | Function) =>
-  component instanceof Function ? component.name : component;
+export const decoratedStory: DecoratedStory = (...args: any[]) => {
+  const [name, context] = args.length === 2 ? args : [args[0].name, args[0]];
```
Comment (codecaaron) [inline packages/styleguide/stories/Templating/DecoratedStory.ts:7]: Aren't most of these functions that we are passing anonymous and won't have function names that we can pull out?

Re: `const blahStory = decoratedStory(() => {});`

Comment (JoshuaKGoldberg) [reply packages/styleguide/stories/Templating/DecoratedStory.ts:7]: Hmm, yes, but it still works anyway _(lol)_, so Storybook must be pulling from variable names too.

File: packages/styleguide/stories/Templating/DecoratedStory.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/styleguide/stories/Templating/DecoratedStory.ts:4]: Yeah this isn't my favorite thing I've ever done... Also, putting the name after context means the name goes _after_ the function, which I found to be kinda unreadable. Thoughts?

Comment (codecaaron) [reply packages/styleguide/stories/Templating/DecoratedStory.ts:4]: Agreed, it is less readable. :shipit:

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.0.1-alpha.45e0db.0<br/>@codecademy/gamut@6.1.0-alpha.45e0db.0<br/></pre>

File: packages/styleguide/stories/Templating/DecoratedStory.ts (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/styleguide/stories/Templating/DecoratedStory.ts:7]: Oh LOL @codecaaron  😄 turns out the opposite part, name overrides, weren't working - they needed to be a sibling of `option`. Changed.

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e9de46a6d53d839c1057e4d--codecademy-styleguide.netlify.app>https://5e9de46a6d53d839c1057e4d--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e9de46a6d53d839c1057e4d>Deploy Logs</a></p>

**Pull request merged**